lib/ace/keyboard/textinput.js

```diff
@@ -328,6 +328,10 @@ var TextInput = function(parentNode, host) {
     var onInput = function(e) {
         if (inComposition)
             return onCompositionUpdate();
+        if (e && e.inputType) {
+            if (e.inputType == "historyUndo") return host.execCommand("undo");
```

Comment:
I have no idea how this interacts with the existing undo/redo code, but ok.

Reply:
Existing undo stops calls preventDefault on keydown, so input event is not emitted.